lib/Dialect/FIRRTL/Transforms/EmitOMIR.cpp

```diff
@@ -279,7 +282,13 @@ void EmitOMIRPass::runOnOperation() {
         makeTrackerAbsolute(tracker);
       trackers.insert({tracker.id, tracker});
       return true;
-    });
+    };
+    AnnotationSet::removePortAnnotations(
+        op, [&](unsigned int portNo, Annotation anno) {
+          return setTracker(anno, portNo);
+        });
+    AnnotationSet::removeAnnotations(
+        op, [&](Annotation anno) { return setTracker(anno); });
```

Comment:
The double lambda wrapping is unfortunate. This could probably cleaned up with something along the lines of (with an import of `<functional>` and `std::placeholders`):

```suggestion
    AnnotationSet::removePortAnnotations(op, std::bind(setTracker, _2, _1));
    AnnotationSet::removeAnnotations(std::bind(setTracker, _1, -1));
```

Reversing the order of arguments in `setTracker` would save the need for the first bind.  Alternatively, you could do the same thing with two lambdas:

```c++
auto setTrackerPort = [&](int portNo, Annotation anno) { /* ... */ };
auto setTrakcer = [&](Annotation anno) { return setTrackerPort(-1, anno); };

AnnotationSet::removePortAnnotations(op, setTrackerPort);
AnnotationSet::removeAnnotations(setTracker);
```

Reply:
Thanks, binder is a nice idea. Updated to use it.